packages/internal/src/lib-chainStorage.js

```diff
@@ -56,8 +57,19 @@ harden(assertPathSegment);
 /**
  * Must match the switch in vstorage.go using `vstorageMessage` type
  *
- * @typedef {'get' | 'getStoreKey' | 'set' | 'append' | 'has' |'entries' | 'values' |'size' } StorageMessageMethod
- * @typedef {{key: string, method: StorageMessageMethod, value: string}} StorageMessage
+ * @typedef { 'get' | 'getStoreKey' | 'has' | 'entries' | 'values' |'size' } StorageGetByPathMessageMethod
+ * @typedef { 'set' | 'append' } StorageUpdateEntriesMessageMethod
+ * @typedef {StorageGetByPathMessageMethod | StorageUpdateEntriesMessageMethod } StorageMessageMethod
+ * @typedef { [path: string] } StorageGetByPathMessageArgs
+ * @typedef { [path: string, value?: string] } StorageEntry
+ * @typedef { StorageEntry[] } StorageUpdateEntriesMessageArgs
+ * @typedef {{
+ *   method: StorageGetByPathMessageMethod;
+ *   args: StorageGetByPathMessageArgs;
+ *  } | {
+ *   method: StorageUpdateEntriesMessageMethod;
+ *   args: StorageUpdateEntriesMessageArgs;
+ * }} StorageMessage
```

Comment:
💯  Always great to see better typings.

Reply:
Well to be fair, the previous typing was 100% correct before I changed the shape of the messages ;)

But yeah I prefer having as descriptive types as possible.